src/types/bool.rs

```diff
@@ -0,0 +1,45 @@
+use crate::ion_data::{IonEq, IonOrd};
+use std::cmp::Ordering;
+use std::fmt::{Display, Formatter};
+
+/// Represents an Ion `bool` value, for the purpose of implementing some Ion-related traits.
+///
+/// Most of the time, you can use [bool] instead of this type.
+#[derive(Debug, Clone, PartialOrd, Ord, PartialEq, Eq)]
+pub struct Bool(bool);
```

Comment:
Probably deriving `Copy` is fine, but I am curious why we need this because we only need to define `IonEq`/`IonOrd` for `bool` (which we should be able to do in the module that defines those traits).  Whatever I am missing from why this is needed probably needs to go in the doc comment.